gitlab/http_backends/requests_backend.py

```diff
@@ -4,6 +4,30 @@
 from requests_toolbelt.multipart.encoder import MultipartEncoder  # type: ignore
 
 
+class RequestsResponse:
+    def __init__(self, response: requests.Response) -> None:
+        self._response: requests.Response = response
+
+    @property
+    def response(self) -> requests.Response:
+        return self._response
+
+    @property
+    def status_code(self) -> int:
+        return self._response.status_code
+
+    @property
+    def headers(self) -> Any:
```

Comment:
Should this be requests' `CaseInsensitiveDict` here?

```suggestion
    def headers(self) -> CaseInsensitiveDict[str]:
```

Reply:
Originally I used it but it caused some errors which I didn't know how to fix. Could you please take a look?

Reply:
I've also filed an issue psf/requests#6311

Reply:
Ah ok, I'll take a look a bit later, thanks. I don't think the requests project can help with this as they don't provide any typing at all themselves, this would be in the typeshed project most likely in case there is an upstream issue.

Reply:
The `requests` project maintainer provided me a hint on the root cause. Pushed a change which removed the errors.